pinot-broker/src/main/java/org/apache/pinot/broker/api/AccessControl.java

```diff
@@ -47,4 +47,13 @@ default boolean hasAccess(RequesterIdentity requesterIdentity) {
    * @return {@code true} if authorized, {@code false} otherwise
    */
   boolean hasAccess(RequesterIdentity requesterIdentity, BrokerRequest brokerRequest);
+
+  /**
+   * Determine whether authentication is required for annotated (controller) endpoints only
+   *
+   * @return {@code true} if annotated methods are protected only, {@code false} otherwise
+   */
+  default boolean protectAnnotatedOnly() {
+    return true;
+  }
```

Comment:
no need for this one here. it was introduced for backwards-compatibility in controller only.

Reply:
+1

Reply:
You should be using @ManualAuthorization annotation for those endpoints, so AuthFilter will simply return with no actual authentication done.